staging/src/k8s.io/apiserver/pkg/server/signal.go

```diff
@@ -67,3 +70,35 @@ func RequestShutdown() bool {
 
 	return false
 }
+
+func dumpStacks(writeToFile bool, path string) {
+	var (
+		buf       []byte
+		stackSize int
+	)
+	bufferLen := 16384
+	for stackSize == len(buf) {
+		buf = make([]byte, bufferLen)
+		stackSize = runtime.Stack(buf, true)
+		bufferLen *= 2
+	}
+	buf = buf[:stackSize]
+	if writeToFile {
+		f, err := os.Create(path)
+		if err != nil {
+			klog.Errorf("failed to create %s err is:%v", path, err)
+			return
+		}
+
+		defer func() {
+			_ = f.Close()
+		}()
+
+		_, err = f.WriteString(string(buf))
```

Comment:
Maybe this can be simplified by using `os.WriteFile` ? 

The conversation in `string(buf)` will allocate memory for string and then copy buffer to that, which can be inefficient.